Added basic implementation for string and path manipulation in python

k-dominik opened the pull request:
This is more or less a 1:1 translation of the macro code to python.

* For log messages: I included an example of using an f-string. While this is an advanced concept It adds so much to the readability of these strings, and also avoids the explicit type conversion.
* For the paths I went for `pathlib.Path`. In principle one could do the same with strings and something like `os.path.join()` to get platform-independent joining of paths (one could also use `os.sep` with string concatenation to be even closer to the macro example. While this would work it encourages bad practices. `pathlib.Path` objects add so much useful functionality (e.g. `path.name` to get the filename, `path.parent` to get the parent directory, `.stem`, `.suffix`... so much that I missed when doing a ij macro).

tischi commented on _includes/string_concat/creating_log_messages.py:
```diff
@@ -1 +1,13 @@
+image_index = 11
+num_images = 100
+
+# strings can be added together with the '+' operator
```
Our convention is to have a `# %%` comment where a new "teaching part" starts.
Could you please add them in the appropriate places? 
Here I would say, in front of all the comment "sections".
Thanks!

tischi commented on _includes/string_concat/manipulating_file_paths.py:
```diff
@@ -1 +1,14 @@
+import pathlib
+import tempfile
+
+# Temporary folder is different on different operating systems.
```
Good to mention what the purpose of this is, something like:

`# Let's get a folder and a file name to practice with`

tischi commented on _includes/string_concat/manipulating_file_paths.py:
```diff
@@ -1 +1,14 @@
+# pathlib.Path objects support "syntactic sugar", like using '/' as a
+# separator when composing paths.
+path = tmp_folder / file_name
+print(path)
```
Could you please add more here:

- how to decompose the path again into `parent` and file name with and without suffix. 
- it would be also important to change the suffix to `.csv` or `_labels.tiff` to be prepared to save derived tables or segmentations.

I guess the reason you did you do this here is that it does not occur in the IJ-Macro implementation? If not (I did not check) would you mind also adding it there?

tischi reviewed (commented):
Thanks a lot, I added a couple of suggestions.

k-dominik commented:
Hello @tischi,

I updated the directory exercise with your proposed changes, both in python and ijm. For the ijm I also removed the used of `File.separator` since the directory we're working with there, already has one. If you find it important didactically to introduce it we'd have to come up with a different way of getting a directory to work with.

tischi commented:
Very cool! 

💟
